test_bst.py

```diff
@@ -78,3 +88,43 @@ def test_post_order(full_tree):
 def test_breadth_first(full_tree):
     g = full_tree.breadth_first()
     assert list(g) == [15, 7, 16, 1, 14, 5]
+
+
+def test_delete_leaf(full_tree):
+    full_tree.delete(16)
+    assert full_tree.size() == 5
+    g = full_tree.in_order()
+    assert list(g) == [1, 5, 7, 14, 15]
+    assert full_tree.contains(16) is False
+
+
+def test_delete_node_with_one_right_child(full_tree):
+    full_tree.delete(1)
+    assert full_tree.size() == 5
+    g = full_tree.in_order()
+    assert list(g) == [5, 7, 14, 15, 16]
+    assert full_tree.contains(1) is False
+
+
+def test_delete_node_with_one_left_child(second_full_tree):
+    second_full_tree.delete(5)
+    assert second_full_tree.size() == 5
+    g = second_full_tree.in_order()
+    assert list(g) == [3, 7, 14, 15, 16]
+    assert second_full_tree.contains(5) is False
+
+
+def test_delete_node_with_two_child(full_tree):
+    full_tree.delete(7)
+    assert full_tree.size() == 5
+    g = full_tree.in_order()
+    assert list(g) == [1, 5, 14, 15, 16]
+    assert full_tree.contains(7) is False
+
+
+def test_delete_root_node(full_tree):
+    full_tree.delete(15)
+    assert full_tree.size() == 5
+    g = full_tree.in_order()
+    assert list(g) == [1, 5, 7, 14, 16]
+    assert full_tree.contains(15) is False
```

Comment:
These tests are good... But they are all just duplicates of themselves. They aren't really testing anything new.. I suggest scrapping all but one. And add tests that check for edge cases, try and break it, and try and touch different angles of the function.